src/Controls/src/Core/Handlers/Items/Android/RecyclerViewScrollListener.cs

```diff
@@ -55,23 +55,42 @@ public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
 
 			_itemsView.SendScrolled(itemsViewScrolledEventArgs);
 
-			// Don't send RemainingItemsThresholdReached event for non-linear layout managers
-			// This can also happen if a layout pass has not happened yet
-			if (Last == -1)
+			// Early returns for edge cases
+			if (Last == -1 || ItemsViewAdapter == null || _itemsView.RemainingItemsThreshold == -1)
+			{
+				return;
+			}
+
+			var itemsSourceCount = ItemsViewAdapter.ItemCount;
+			bool hasHeader = ItemsViewAdapter.ItemsSource.HasHeader;
+			bool hasFooter = ItemsViewAdapter.ItemsSource.HasFooter;
+
+			int firstDataItemIndex = (hasHeader && hasFooter) ? 1 : 0;
+			int lastDataItemIndex = itemsSourceCount - firstDataItemIndex;
+
+			if (Last < firstDataItemIndex || Last > lastDataItemIndex)
+			{
 				return;
+			}
 
 			switch (_itemsView.RemainingItemsThreshold)
```

Comment:
Here, could simplify the logic by consolidating checks:

```
bool isThresholdReached = (Last == lastDataItemIndex - 1) || (lastDataItemIndex - 1 - Last <= _itemsView.RemainingItemsThreshold);

if (isThresholdReached)
{
    _itemsView.SendRemainingItemsThresholdReached();
}
```

Reply:
@jsuarezruiz , I've optimized the logic for improved clarity and efficiency. Please review the modified fix and let me know if any further changes are required.